### PR TITLE
Rental revisited

### LINKED ISSUE #172: rental revisited
# show
* change status to be human readable (remove under score)
* remove date range from title
* rearrange fields
  * billing information at the top
  * details at the bottom
* rename payment summary to financial transactions
  * payments should be negative values (may be the current behavior)

# index
* display canceled rentals in a readable color

# transform (process)
* check out
  * add pick up customer information (text field)
  * ability to change contact information
  * default fully charged and confirmed return time to yes
* check in
  * add drop off customer information (text field)
  * overdue rentals should actually be checked it, right now the form does nothing
  * report incurred incidentals
# general
* add fields to mode
  * reserved by (rename user name)
  * picked up by (text field)

### LINKED ISSUE #173: home page
* move 'all rentals' to last tab, 'upcoming rentals' should be default
* constant column width

### PR DESCRIPTION
**Pull request by Cobchise**
### Closes #172 Closes #173

- Implements many small style changes

- Added customer information (name and phone number) to transformation process (check in/ check out)

- Fixed bug that prevented overdue rentals from being checked in (transitioning from reserved to canceled). Bug was caused by validations conflicting with the start date and not allowing rental to be saved. Fixed by saving start and end date of rental to day of cancelation. (Database still holds date of creation of rental if needed)

Addresses all points in issues except  "report incurred incidentals on check in". We discussed how we would approach this and have not yet reached a decision. Ideas were:

- A pop out window approach 

- Adding incidentals before check in manually

- Integrating adding incurred incidentals with check in form

### CODE REVIEW

File: app/views/rentals/show.html.erb [hunk 1]
```diff
@@ -26,67 +41,76 @@
...
   <label class="col-xs-2 control-label">Department</label>
   <div class="form-control-informaton col-xs-10"><%= @rental.department.name %></div>
 </div>
 
+<% if @rental.pickup_name.present? %>
 <div class="form-group col-xs-12">
-  <label class="col-xs-2 control-label">Amount Charged</label>
-  <div class="form-control-information col-xs-10">$</div>
+  <label class="col-xs-2 control-label">Picked Up By:</label>
+  <div class="form-control-informaton col-xs-10"><%= @rental.pickup_name + " (Phone: #{number_to_phone(@rental.pickup_phone_number.gsub(/[\s+)(-]/, ""), area_code: true)})" %></div>
```
Comment (tomecho) [inline app/views/rentals/show.html.erb]: please move logic to a helper file

File: app/views/home/index.html.erb [hunk 2]
```diff
@@ -4,12 +4,12 @@
...
-    <li><a data-toggle="tab" href="#future">All Future Rentals</a></li>
+    <li class="active"><a data-toggle="tab" href="#upcoming">Upcoming Rentals <span class="badge"><%= @upcoming_rentals.count %></span></a></li>
     <li><a data-toggle="tab" href="#ongoing">Ongoing Rentals <span class="badge"><%= @ongoing_rentals.count %></span></a></li>
     <li><a data-toggle="tab" href="#noshow">No-Show Rentals <span class="badge"><%= @no_show_rentals.count %></span></a></li>
     <li><a data-toggle="tab" href="#past">Past Rentals</a></li>
+    <li><a data-toggle="tab" href="#future">All Future Rentals</a></li>
+    <li><a data-toggle="tab" href="#all">All Rentals</a></li>
   </ul>
 
   <div class="tab-content">
```
Comment (tomecho) [inline app/views/home/index.html.erb]: tab content needs to be modified too

Comment (tomecho) [inline app/views/home/index.html.erb]: also it has really bad indentation

Comment (Cobchise) [inline app/views/home/index.html.erb]: Fixed 👍

**Pull request merged**